addons/l10n_my_edi/models/account_move.py

```diff
@@ -0,0 +1,661 @@
+        error = ''
+        # The api returns the result per document uuid, already correctly formated.
+        # We do not need to format the data anymore for processing later but just to ensure we get complete data of the submission.
+        result = proxy_user._l10n_my_edi_contact_proxy(
+            endpoint='api/l10n_my_edi/1/get_submission_statuses',
+            params={
+                'submission_uid': submission_uid,
+                'page': 1,
+            }
+        )
+
+        if 'error' in result:
+            error = self._l10n_my_edi_map_error(result['error'])
+        else:
+            if result['document_count'] <= 100:  # If so, we got all of it at once.
+                result = result['statuses']
+            else:
+                # Otherwise we'll need to get the remaining invoices per batch of 100.
+                for page in range(2, (result['document_count'] // 100) + 1):
+                    time.sleep(1)  # To avoid any risks of throteling, we should wait a bit before continuing
+                    page_result = proxy_user._l10n_my_edi_contact_proxy(
+                        endpoint='api/l10n_my_edi/1/get_submission_statuses',
+                        params={
+                            'submission_uid': submission_uid,
+                            'page': page,
+                        }
+                    )
+                    result['statuses'].update(page_result['statuses'])
+                result = result['statuses']
+        return error, result
+
+    def _l10n_my_edi_fetch_status(self):
+        """ Action to fetch the status of a single invoice. """
+        self.ensure_one()
+        proxy_user = self._l10n_my_edi_ensure_proxy_user()
+
+        # What to do with the given status is to be handled by the calling code.
+        return proxy_user._l10n_my_edi_contact_proxy(
+            endpoint='api/l10n_my_edi/1/get_status',
+            params={
+                'document_uuid': self.l10n_my_edi_external_uuid,
+            },
+        )
+
+    # Other methods
+
+    def _l10n_my_edi_uses_edi(self):
+        """ Helper that returns true if the invoices company is using the Malaysian EDI.
+        It does not mean that this specific invoice will use it, though.
+        """
+        self.ensure_one()
+        proxy_user = self.company_id.l10n_my_edi_proxy_user_id
+        return proxy_user and proxy_user.proxy_type == 'l10n_my_edi'
+
+    def _l10n_my_edi_generate_invoice_xml(self):
+        """ This edi's file is basically an ubl 2.1 file with some specificities. """
+        self.ensure_one()
+        return self.env['account.edi.xml.ubl_myinvois_my']._export_invoice(self)
+
+    def _l10n_my_edi_ensure_proxy_user(self):
+        # We need this fallback, as this method could be called by a cron.
+        company = self.company_id or self.env.company
+
+        proxy_user = company.l10n_my_edi_proxy_user_id
```

Comment:
What if the proxy user is deactivated? should we still use it to send/retrieve invoices?

Reply:
I believe company.l10n_my_edi_proxy_user_id will return an empty recordset if the user is deactivated, and thus it will not show the option to send in the send & print/will raise.
It may not be perfect, but the only use case where this should happen would be duplicate databases where we don't want it to work anyway.

Reply:
You're right about that! Never mind :smile: